### PR TITLE
Tests rewards

### PR DESCRIPTION
**Pull request by AboudyKreidieh**
changes:
- bug fix to `penalize_near_standstill`
- expanded unit test coverage for `average_velocity`
- unit tests for `max_edge_velocity`, `min_delay`, `penalize_standstill`, and `penalize_near_standstill`

### CODE REVIEW

**Comment by kjang96** [comment]
dibs

File: flow/core/rewards.py [hunk 1]
```diff
@@ -21,14 +21,22 @@ def desired_velocity(env, fail=False):
...
         specifies if any crash or other failure occurred in the system
+    edge_list : list  of str, optional
+        list of edges the reward is computed over. If no edge_list is defined,
+        the reward is computed over all edges
     """
-    vel = np.array(env.vehicles.get_speed(env.vehicles.get_ids()))
-    num_vehicles = env.vehicles.num_vehicles
+    if edge_list is None:
+        veh_ids = env.vehicles.get_ids()
+    else:
```
Comment (eugenevinitsky) [inline flow/core/rewards.py]: Good call! This is useful.

File: flow/core/rewards.py [hunk 2]
```diff
@@ -202,7 +168,7 @@ def penalize_standstill(env, gain=1):
 def penalize_near_standstill(env, thresh=0.3, gain=1):
     veh_ids = env.vehicles.get_ids()
     vel = np.array(env.vehicles.get_speed(veh_ids))
-    penalize = len(vel[vel < 0.3])
+    penalize = len(vel[vel < thresh])
```
Comment (eugenevinitsky) [inline flow/core/rewards.py]: Good catch.

File: tests/fast_tests/test_rewards.py [hunk 3]
```diff
@@ -29,12 +32,27 @@ def test_desired_velocity(self):
...
 
+        # check the average speed upon reset with a subset of edges
+        self.assertEqual(desired_velocity(env, edge_list=["bottom"],
+                                          fail=False), 0)
+
         # change the speed of one vehicle
         env.vehicles.test_set_speed("test_0", 10)
 
         # check the new average speed
         self.assertAlmostEqual(desired_velocity(env, fail=False), 0.05131670)
```
Comment (eugenevinitsky) [inline tests/fast_tests/test_rewards.py]: Can you pick a number such that this is basically an integer? For example, put just one or two vehicles into the system. It just seems cleaner and then obvious when something is wrong.

File: tests/fast_tests/test_rewards.py [hunk 4]
```diff
@@ -29,12 +32,27 @@ def test_desired_velocity(self):
...
 
+        # check the new average speed for a subset of edges
+        self.assertAlmostEqual(desired_velocity(env, edge_list=["bottom"],
+                                                fail=False), 0.18350341907)
```
Comment (eugenevinitsky) [inline tests/fast_tests/test_rewards.py]: As above

File: tests/fast_tests/test_rewards.py [hunk 5]
```diff
@@ -29,12 +32,27 @@ def test_desired_velocity(self):
...
+
+        # change the speed of one of the vehicles outside the edge list
+        env.vehicles.test_set_speed("test_8", 10)
+
+        # check that the new average speed is the same as before
+        self.assertAlmostEqual(desired_velocity(env, edge_list=["bottom"],
+                                                fail=False), 0.18350341907)
```
Comment (eugenevinitsky) [inline tests/fast_tests/test_rewards.py]: As above

File: flow/envs/green_wave_env.py [hunk 6]
```diff
@@ -226,7 +226,8 @@ def _apply_rl_actions(self, rl_actions):
 
     def compute_reward(self, rl_actions, **kwargs):
         """See class definition."""
-        return rewards.penalize_tl_changes(rl_actions >= 0.5, gain=1.0)
+        # penalize traffic light changes for occurring
```
Comment (eugenevinitsky) [inline flow/envs/green_wave_env.py]: Do you have any thoughts on how we can make this nicer with the threshold? It seems like this is really ripe for mistake i.e. if your env doesn't use 0.5 as its switching threshold you'll accidentally create a bug.

**Pull request merged**